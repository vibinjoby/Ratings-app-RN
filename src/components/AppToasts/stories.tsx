import React from 'react'
import { StyleSheet } from 'react-native'
import { storiesOf } from '@storybook/react-native'

import AppToasts from './component'

storiesOf('Components', module).add('AppToasts', () => (
  <AppToasts isError={false} toastMessage="Hey this is a toast" />
))

const styles = StyleSheet.create({
  buttons: {
    margin: 20,
  },
})
