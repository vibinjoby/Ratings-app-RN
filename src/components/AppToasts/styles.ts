import { StyleSheet } from 'react-native'
import Colors from '../../utilities/colors'

export default StyleSheet.create({
  centeredView: {
    flex: 1,
    position: 'absolute',
    bottom: 0,
    width: '100%',
  },
  err: { borderLeftColor: Colors.red },
  animatableView: {
    position: 'absolute',
    bottom: 0,
    width: '100%',
  },
  modalView: {
    margin: 20,
    backgroundColor: Colors.white,
    borderLeftColor: Colors.green,
    borderLeftWidth: 8,
    borderRadius: 8,
    padding: 20,
    shadowColor: Colors.black,
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    elevation: 5,
  },
  modalViewContent: {
    flexDirection: 'row',
    flex: 1,
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  modalText: {
    textAlign: 'center',
    fontSize: 16,
    alignSelf: 'center',
    width: '60%',
    flexWrap: 'wrap',
  },
  toastImg: {
    width: 30,
    height: 30,
  },
  closeToast: {
    marginTop: 10,
    width: 12,
    height: 12,
  },
})
