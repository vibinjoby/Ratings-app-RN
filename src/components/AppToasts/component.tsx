import React, { useState, useEffect } from 'react'
import * as Animatable from 'react-native-animatable'
import { View, Text, Image, TouchableOpacity, TouchableWithoutFeedback } from 'react-native'
import Modal from 'react-native-modal'
import { useDispatch, useSelector } from 'react-redux'

import { RootState } from '../../store'
import { resetToastMessage } from '../../store/reducers/common/toaster'
import styles from './styles'

const AppToasts: React.FC = ({ children }: React.Props<any>) => {
  const toaster = useSelector((state: RootState) => state.toaster)
  const [modalVisible, setModalVisible] = useState(true)
  const [animationType, setAnimationType] = useState('slideInRight')
  const dispatch = useDispatch()

  useEffect(() => {
    if (!toaster.toastMessage) return
    setModalVisible(true)
    // After 3 sec close the toast using timeout animation to right
    setTimeout(() => {
      hideToaster()
    }, 3000)
    setAnimationType('slideInRight')
  }, [toaster])

  const hideToaster = () => {
    setAnimationType('slideOutRight')
    setTimeout(() => {
      dispatch({ type: resetToastMessage.type })
      setModalVisible(false)
    }, 800)
  }

  if (!toaster?.toastMessage) return <>{children}</>

  return (
    <>
      {children}
      <Modal isVisible={modalVisible} onBackdropPress={hideToaster} backdropOpacity={0.3}>
        <Animatable.View animation={animationType} style={styles.animatableView}>
          <TouchableOpacity
            style={styles.centeredView}
            onPress={() => {
              setModalVisible(false)
            }}
          >
            <View style={[styles.modalView, !toaster.isSuccess && { borderLeftColor: 'red' }]}>
              <View style={styles.modalViewContent}>
                <Image
                  source={
                    toaster.isSuccess
                      ? require('../../assets/success-toasts.png')
                      : require('../../assets/error-toasts.png')
                  }
                  resizeMode="contain"
                  style={styles.toastImg}
                />
                <Text style={styles.modalText}>{toaster.toastMessage}</Text>
                <TouchableWithoutFeedback onPress={() => setModalVisible(false)}>
                  <Image
                    source={require('../../assets/close-toasts.png')}
                    resizeMode="contain"
                    style={styles.closeToast}
                  />
                </TouchableWithoutFeedback>
              </View>
            </View>
          </TouchableOpacity>
        </Animatable.View>
      </Modal>
    </>
  )
}

export default AppToasts
