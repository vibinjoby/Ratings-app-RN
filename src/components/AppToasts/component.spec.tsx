import 'react-native'
import React from 'react'
import { render } from '@testing-library/react-native'

import AppToasts from './component'

describe('AppToasts', () => {
  beforeEach(() => {
    jest.useFakeTimers()
  })

  it('should render', () => {
    const { toJSON } = render(<AppToasts isError={false} toastMessage="Toast message" />)
    expect(toJSON()).toMatchSnapshot()
  })

  it('should show toast message', () => {
    const { getByText } = render(<AppToasts isError={false} toastMessage="Toast message" />)

    getByText('Toast message')
  })
})
