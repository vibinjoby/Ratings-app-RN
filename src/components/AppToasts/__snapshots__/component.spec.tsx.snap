// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppToasts should render 1`] = `
<Modal
  animationType="none"
  deviceHeight={null}
  deviceWidth={null}
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onBackdropPress={[Function]}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  panResponderThreshold={4}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  statusBarTranslucent={false}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    focusable={true}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    deviceHeight={null}
    deviceWidth={null}
    forwardedRef={[Function]}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    panResponderThreshold={4}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    statusBarTranslucent={false}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "margin": 37.5,
        "transform": Array [
          Object {
            "translateY": 1334,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <View
      forwardedRef={[Function]}
      style={
        Object {
          "bottom": 0,
          "position": "absolute",
          "transform": Array [
            Object {
              "translateX": 750,
            },
          ],
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "bottom": 0,
            "flex": 1,
            "opacity": 1,
            "position": "absolute",
            "width": "100%",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#fff",
                "borderLeftColor": "green",
                "borderLeftWidth": 8,
                "borderRadius": 8,
                "elevation": 5,
                "margin": 20,
                "padding": 20,
                "shadowColor": "#181725",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 3.84,
              },
              false,
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Image
              resizeMode="contain"
              source={
                Object {
                  "testUri": "../../../src/assets/success-toasts.png",
                }
              }
              style={
                Object {
                  "height": 30,
                  "width": 30,
                }
              }
            />
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "flexWrap": "wrap",
                  "fontSize": 16,
                  "textAlign": "center",
                  "width": "60%",
                }
              }
            >
              Toast message
            </Text>
            <Image
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              resizeMode="contain"
              source={
                Object {
                  "testUri": "../../../src/assets/close-toasts.png",
                }
              }
              style={
                Object {
                  "height": 12,
                  "marginTop": 10,
                  "width": 12,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
